src/Workload/Microsoft.Maui.Sdk/Sdk/BundledVersions.in.targets

```diff
@@ -19,49 +20,88 @@
       <PrivateAssets Condition=" '$(OutputType)' == 'Library' and '$(AndroidApplication)' != 'true' ">all</PrivateAssets>
     </_MauiImplicitPackageReference>
-  </ItemGroup>
-  <ItemGroup Condition=" '$(UseMaui)' == 'true' and '$(UseMauiCompat)' == 'true' ">
-    <_MauiImplicitPackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="$(MauiVersion)">
+    <_MauiImplicitPackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="$(MauiVersion)" IsMauiImplicitOptionalPackageReference="true" Condition=" '$(UseMaui)' == 'true' and '$(UseMauiCompat)' == 'true' ">
       <PrivateAssets Condition=" '$(OutputType)' == 'Library' and '$(AndroidApplication)' != 'true' ">all</PrivateAssets>
     </_MauiImplicitPackageReference>
-  </ItemGroup>
-  <ItemGroup Condition=" '$(UseMaui)' == 'true' and '$(UsingMicrosoftNETSdkRazor)' == 'true' ">
-    <_MauiImplicitPackageReference Include="Microsoft.AspNetCore.Components.WebView.Maui" Version="$(MauiVersion)">
+    <_MauiImplicitPackageReference Include="Microsoft.AspNetCore.Components.WebView.Maui" Version="$(MauiVersion)" Condition=" '$(UseMaui)' == 'true' and '$(UsingMicrosoftNETSdkRazor)' == 'true' ">
       <PrivateAssets Condition=" '$(OutputType)' == 'Library' and '$(AndroidApplication)' != 'true' ">all</PrivateAssets>
     </_MauiImplicitPackageReference>
   </ItemGroup>
-  <!-- Always install the required dependecies, such as build tools for the SDK itself -->
-  <ItemGroup>
-    <_MauiRequiredImplicitPackageReference Remove="@(PackageReference)" />
-    <PackageReference Include="@(_MauiRequiredImplicitPackageReference)" />
-  </ItemGroup>
-  <!-- Install the nuget packages if we are supposed to -->
-  <ItemGroup Condition=" '$(UseMauiNuGets)' != 'true' ">
+  <!-- Install the nuget packages if we are allowed to -->
+  <ItemGroup Condition=" '$(EnableMauiImplicitPackageReferences)' != 'false' and '$(UseMauiNuGets)' != 'true' ">
     <_MauiImplicitPackageReference Remove="@(PackageReference)" />
-    <PackageReference Include="@(_MauiImplicitPackageReference)" />
+    <PackageReference Include="@(_MauiImplicitPackageReference)" IsMauiImplicitPackageReference="true" />
+    <_MauiImplicitRequiredPackageReference Include="@(_MauiImplicitPackageReference->WithMetadataValue('IsMauiImplicitOptionalPackageReference',''))" />
   </ItemGroup>
 
+  <Target
+    Name="ValidateMauiImplicitPackageReferences"
+    AfterTargets="CollectPackageReferences"
+    Condition="
+      ('$(SkipValidateMauiImplicitPackageReferences)' != 'true') and
+      ('$(EnableMauiImplicitPackageReferences)' != 'false' and '$(UseMauiNuGets)' != 'true') and
+      ('$(UseMaui)' == 'true' or '$(UseMauiCore)' == 'true' or '$(UseMauiEssentials)' == 'true' or '$(UseMauiAssets)' == 'true') and
+      ('@(_MauiImplicitRequiredPackageReference->Count())' != '0') ">
+
+    <PropertyGroup>
+      <_MauiValidateRefsPrefix>Starting with .NET 8, setting </_MauiValidateRefsPrefix>
```

Comment:
These messages... @rmarinho what is the way to localize?

Reply:
Use resources? how do you pull into here I think I saw something about that

Reply:
@mcumming @pjcollins do you know?

Reply:
Here's how the dotnet teams does it 

https://github.com/dotnet/sdk/blob/880798d13942d41e61201e4281604560eade31e9/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.EolTargetFrameworks.targets#L28

Reply:
https://github.com/dotnet/sdk/blob/880798d13942d41e61201e4281604560eade31e9/src/Tasks/Common/MessageBase.cs#L63

Reply:
I created an issue for the localization part: https://github.com/dotnet/maui/issues/14804

I found a few plain text strings that we need to localize across our targets.